velox/substrait/tests/Substrait2VeloxPlanConversionTest.cpp

```diff
@@ -286,3 +287,137 @@ TEST_F(Substrait2VeloxPlanConversionTest, q6) {
+  ::substrait::Plan substraitPlan;
+  JsonToProtoConverter::readFromFile(planPath, substraitPlan);
+
+  facebook::velox::substrait::SubstraitVeloxPlanConverter planConverter(
+      pool_.get());
+  auto planNode = planConverter.toVeloxPlan(substraitPlan);
+  // Note: input data to this test comes from a virtual table in the plan
+  auto expectedResult = makeRowVector(
+      {makeFlatVector<int64_t>({1000, 2000, 3000}),
+       makeFlatVector<double_t>({0.2, 0.5, 0.4}),
+       makeFlatVector<bool>({true, false, false})});
+
+  exec::test::AssertQueryBuilder(planNode).assertResults(expectedResult);
+}
+
+TEST_F(
+    Substrait2VeloxPlanConversionTest,
+    NestedProjectWithMultiFieldExpressions) {
+  std::string planPath = getDataFilePath(
+      "velox/substrait/tests", "data/substrait_multi_project_emit.json");
+  ::substrait::Plan substraitPlan;
+  JsonToProtoConverter::readFromFile(planPath, substraitPlan);
+
+  facebook::velox::substrait::SubstraitVeloxPlanConverter planConverter(
+      pool_.get());
+  auto planNode = planConverter.toVeloxPlan(substraitPlan);
+  // Note: input data to this test comes from a virtual table in the plan
+  auto expectedResult = makeRowVector(
+      {makeFlatVector<double_t>({100, 200, 20}),
+       makeFlatVector<double_t>({100, 200, 20})});
+  exec::test::AssertQueryBuilder(planNode).assertResults(expectedResult);
+}
+
+TEST_F(
+    Substrait2VeloxPlanConversionTest,
+    NestedEmitProjectWithMultiFieldExpressions) {
+  std::string planPath = getDataFilePath(
+      "velox/substrait/tests", "data/substrait_nested_multi_project_emit.json");
+  ::substrait::Plan substraitPlan;
+  JsonToProtoConverter::readFromFile(planPath, substraitPlan);
+
+  facebook::velox::substrait::SubstraitVeloxPlanConverter planConverter(
+      pool_.get());
+  auto planNode = planConverter.toVeloxPlan(substraitPlan);
+  // Note: input data to this test comes from a virtual table in the plan
+  auto expectedResult =
+      makeRowVector({makeFlatVector<int64_t>({1000, 2000, 3000})});
+  exec::test::AssertQueryBuilder(planNode).assertResults(expectedResult);
+}
+
+TEST_F(Substrait2VeloxPlanConversionTest, ReadRelWithEmit) {
+  facebook::velox::substrait::SubstraitTestPlanBuilder builder;
+  auto c0 = makeFlatVector<int64_t>({1000, 2000, 3000});
+  auto c1 = makeFlatVector<double>({10.0, 20.0, 2.0});
+  auto c2 = makeFlatVector<double>({0.2, 0.5, 0.4});
+  auto c3 = makeFlatVector<bool>({true, false, false});
+  auto c4 = makeFlatVector<int64_t>({10, 12, 11});
+
+  auto data =
+      makeRowVector({"c0", "c1", "c2", "c3", "c4"}, {c0, c1, c2, c3, c4});
+
+  auto expectedResult = makeRowVector({
+      makeFlatVector<int64_t>({1000, 2000, 3000}),
+      makeFlatVector<double_t>({0.2, 0.5, 0.4}),
+      makeFlatVector<bool>({true, false, false}),
+  });
+
+  builder.ValidateReadRelWithEmit(
+      {data}, /*expected*/ {expectedResult}, /*emitIndices*/ {0, 2, 3});
+}
```

Comment:
@mbasmanova I updated the code not to use JSON, but I am not sure if this method meets expectations. 
The helper classes are `SubstraitTestPlanBuilder.h` and `SubstraitTestPlanBuilder.cpp`. Please take a look. 

PS: I need to make this change reflected in other tests, but before moving there, I want to validate if this is okay.